tangelo/toolboxes/qubit_mappings/combinatorial.py

```diff
@@ -0,0 +1,234 @@
+                phase *= phase_two
+
+            if not new_state:
+                continue
+
+            new_unique_int = basis_set[new_state]
+            qu_op += element_to_qubitop(n, unique_int, new_unique_int, phase*coeff)
+
+    return qu_op
+
+
+def element_to_qubitop(n_qubits, i, j, coeff=1.):
+    """Map a matrix element to a qubit operator.
+
+    Args:
+        n_qubits (int): Self-explanatory.
+        i (int): i row of the matrix element.
+        j (int): j column of the matrix element.
+        coeff (complex): Value at position i,j in the matrix.
+
+    Returns:
+        QubitOperator: Self-explanatory.
+    """
+
+    # Must add 2 to the padding because of the "0b" prefix.
+    bin_i = format(i, f"#0{n_qubits+2}b")
+    bin_j = format(j, f"#0{n_qubits+2}b")
+
+    qu_op = QubitOperator("", coeff)
+    for qubit, (bi, bj) in enumerate(zip(bin_i[2:][::-1], bin_j[2:][::-1])):
+        if bi == "0" and bj == "0":
+            qu_op *= 0.5 + QubitOperator(f"Z{qubit}", 0.5)
+        elif bi == "0" and bj == "1":
+            qu_op *= QubitOperator(f"X{qubit}", 0.5) + QubitOperator(f"Y{qubit}", 0.5j)
+        elif bi == "1" and bj == "0":
+            qu_op *= QubitOperator(f"X{qubit}", 0.5) + QubitOperator(f"Y{qubit}", -0.5j)
+        # The remaining case is 11.
+        else:
+            qu_op *= 0.5 + QubitOperator(f"Z{qubit}", -0.5)
+
+    return qu_op
+
+
+def basis(M, N):
+    """Function to construct the combinatorial basis set, i.e. a basis set
+    respecting the number of electrons and the total spin.
+
+    Args:
+        M (int): Number of spatial orbitals.
+        N (int): Number of alpha or beta electrons.
+
+    Returns:
+        OrderedDict: Lexicographically sorted basis set, mapping electronic
+            configuration to unique integers.
+    """
+
+    mapping = [(sigma, conf_to_integer(sigma, M)) for sigma in itertools.combinations(range(M), N)]
+    return OrderedDict(mapping)
+
+
+def conf_to_integer(sigma, M):
+    """Function to map an electronic configuration to a unique integer, as done
+    in arXiv.2205.11742 eq. (14).
+
+    Args:
+        sigma (tuple of int): Orbital indices where the electron are in the
+            electronic configuration.
+        M (int): Number of modes, i.e. number of spatial orbitals.
+
+    Returns:
+        int: Unique integer for the input electronic state.
+    """
+
+    # Equivalent to the number of electrons.
+    N = len(sigma)
+
+    # Eq. (14) in the reference.
+    terms_k = [comb(M - sigma[N - 1 - k] - 1, k + 1) for k in range(N)]
+    unique_int = comb(M, N) - 1 - np.sum(terms_k)
+
+    return int(unique_int)
+
+
+def one_body_op_on_state(op, state_in):
+    """Function to apply a a^{\dagger}_i a_j operator as described in Phys. Rev.
+    A 81, 022124 (2010) eq. (8).
+
+    Args:
+        op (tuple): Operator, written as ((qubit_i, 1), (qubit_j, 0)), where 0/1
+            means anhilation/creation on the specified qubit.
```

Comment:
it's annihilation I believe, everywhere.
I suggest automated replace all, because some appears in your code, some in docstrings / comments